tox.ini

```
@@ -59,17 +59,56 @@ commands =
 deps =
     -r{toxinidir}/requirements/quality.txt
 commands =
-    black --check --diff --line-length=80 src/scriv tests docs setup.py
+    black --check --diff --line-length=80 src/scriv tests docs
     python -m cogapp -cP --check --verbosity=1 docs/*.rst
     mypy src/scriv tests
-    pylint src/scriv tests docs setup.py
-    pycodestyle src/scriv tests docs setup.py
-    pydocstyle src/scriv tests docs setup.py
-    isort --check-only --diff -p scriv tests src/scriv setup.py
-    python setup.py -q sdist
+    pylint src/scriv tests docs
+    pycodestyle src/scriv tests docs
+    pydocstyle src/scriv tests docs
+    isort --check-only --diff -p scriv tests src/scriv
+    python -m build --sdist
     twine check dist/*
 
 [testenv:upgrade]
 commands =
     python -m pip install -U pip
     make upgrade
+
+# Tools needed for running tests need to be configured in tox.ini instead of
+# pyproject.toml so that we can run tests without tomli installed on Pythons
+# before 3.11.
+
+[pytest]
+addopts = -rfe
+norecursedirs = .* docs requirements
+
+[coverage:run]
+branch = True
+source =
+    scriv
+    tests
```

I've found that things like Codecov are happier with the following config:
```suggestion
source =
    .
source_pkgs =
    scriv
```

---

I'm not a fan of codecov, and this works.